src/contracts/core/StakeRootCompendium.sol

```diff
@@ -0,0 +1,618 @@
+    }
+
+    function _getStrategiesAndMultipliers(
+        OperatorSet memory operatorSet
+    ) internal view returns (IStrategy[] memory, uint256[] memory) {
+        IStrategy[] memory strategies =
+            new IStrategy[](operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length());
+        uint256[] memory multipliers = new uint256[](strategies.length);
+        for (uint256 i = 0; i < strategies.length; i++) {
+            (address strategy, uint256 multiplier) =
+                operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].at(i);
+            strategies[i] = IStrategy(strategy);
+            multipliers[i] = multiplier;
+        }
+        return (strategies, multipliers);
+    }
+
+    function _getStakes(
+        OperatorSet memory operatorSet,
+        IStrategy[] memory strategies,
+        uint256[] memory multipliers,
+        address operator
+    ) internal view returns (uint256, uint256) {
+        // calculate the weighted sum of the operator's shares for the strategies given the multipliers
+        uint256 delegatedStake = 0;
+        uint256 slashableStake = 0;
+        {
+            uint256[] memory delegatedShares = delegationManager.getOperatorScaledShares(operator, strategies);
+
+            (uint64[] memory totalMagnitudes, uint64[] memory allocatedMagnitudes) =
+                allocationManager.getTotalAndAllocatedMagnitudes(operator, operatorSet, strategies);
+
+            for (uint256 i = 0; i < strategies.length; i++) {
+                delegatedStake += delegatedShares[i] * totalMagnitudes[i] / 1 ether * multipliers[i];
+                slashableStake += delegatedShares[i] * allocatedMagnitudes[i] / 1 ether * multipliers[i];
+            }
+        }
+        return (delegatedStake, slashableStake);
+    }
+
+    function _isInStakeTree(
+        OperatorSet memory operatorSet
+    ) internal view returns (bool) {
+        (bool exists,, uint224 index) = operatorSetToIndex[operatorSet.avs][operatorSet.operatorSetId].latestSnapshot();
+        return exists && index != REMOVED_INDEX;
+    }
+
+    // VIEW FUNCTIONS
+
+    /// VIEW FUNCTIONS
+
+    /// @inheritdoc IStakeRootCompendium
+    function minDepositBalance(
+        uint256 numStrategies
+    ) public view returns (uint256) {
+        ChargeParams memory charges = chargeParams;
+        return (numStrategies * charges.chargePerStrategy + charges.chargePerOperatorSet) * MIN_PREPAID_PROOFS;
+    }
+
+    /// @inheritdoc IStakeRootCompendium
+    function canWithdrawDepositBalance(
+        OperatorSet memory operatorSet
+    ) public view returns (bool) {
+        // they must have paid for all of their prepaid proofs before withdrawing after a demand increase
+        return block.timestamp
+            > depositInfos[operatorSet.avs][operatorSet.operatorSetId].lastDemandIncreaseTimestamp
```

Comment:
What happens if demand decreases (ie. remove strategies/multipliers)? Shouldn't the AVS be on the hook for the smaller set of strategies/multipliers instead of all of them when initially set?